csrc/multidevice/utils.h

```diff
@@ -19,6 +19,9 @@ namespace nvfuser {
 // Checks that the other non-reduction axis are not parallelized on Didx
 NVF_API bool isSharded(TensorView*);
 
+// Returns number of device dimensions in a TensorView's leaf domain.
+NVF_API int64_t numDeviceDims(TensorView*);
```

Comment:
https://github.com/NVIDIA/Fuser/blob/main/doc/dev/visibility.md#should-i-mark-my-new-method-or-class-as-nvf_api

(We can clean it up in a separate PR like https://github.com/NVIDIA/Fuser/pull/2161)